svc-systems/chassis/fabricresource.go

```diff
@@ -34,18 +36,23 @@ var (
 
 // getFabricChassisResource will collect the individual
 // fabric chassis resourse from all the fabric plugin
-func (f *fabricFactory) getFabricChassisResource(rID string) response.RPC {
+func (f *fabricFactory) getFabricChassisResource(ctx context.Context, rID string) response.RPC {
 	var resp response.RPC
 	ch := make(chan response.RPC)
 
-	managers, err := f.getFabricManagers()
+	managers, err := f.getFabricManagers(ctx)
+	l.LogWithFields(ctx).Debugf("Inside getFabricChassisResource")
 	if err != nil {
-		l.Log.Warn("while trying to collect fabric managers details from DB, got " + err.Error())
+		l.LogWithFields(ctx).Warn("while trying to collect fabric managers details from DB, got " + err.Error())
 		return common.GeneralError(http.StatusNotFound, response.ResourceNotFound, "", []interface{}{"Chassis", rID}, nil)
 	}
 
+	threadID := 1
 	for _, manager := range managers {
-		go f.getResource(manager, rID, ch)
+		ctxt := context.WithValue(ctx, common.ThreadName, common.CollectChassisResource)
+		ctx = context.WithValue(ctxt, common.ThreadID, strconv.Itoa(threadID))
```

Comment:
it should be ctxt ..no? and we should pass ctxt in getResource function

Reply:
fixed